File: apps/sdkwork-birdcoder-h5/src/bootstrap/runtime.ts
import {
  normalizeBirdCoderServerBaseUrl,
  resolveBirdCoderBootstrapServerBaseUrl,
} from '@sdkwork/birdcoder-h5-core';

export function createRuntime(options: {
  configuredApiBaseUrl?: string;
  storedApiBaseUrl?: string;
} = {}) {
  const apiBaseUrl = resolveBirdCoderBootstrapServerBaseUrl({
    configuredApiBaseUrl: normalizeBirdCoderServerBaseUrl(options.configuredApiBaseUrl),
    storedApiBaseUrl: normalizeBirdCoderServerBaseUrl(options.storedApiBaseUrl),
  });

  return {
    apiBaseUrl,
  };
}

export {
  normalizeBirdCoderServerBaseUrl,
  resolveBirdCoderBootstrapServerBaseUrl,
};

File: apps/sdkwork-birdcoder-flutter-mobile/lib/bootstrap/runtime.dart
import 'package:sdkwork_birdcoder_flutter_mobile_core/sdkwork_birdcoder_flutter_mobile_core.dart'
    as core;

String resolveBirdCoderBootstrapServerBaseUrl({
  String? configuredApiBaseUrl,
  String? runtimeApiBaseUrl,
  String? storedApiBaseUrl,
}) {
  return core.resolveBirdCoderBootstrapServerBaseUrl(
        configuredApiBaseUrl: configuredApiBaseUrl,
        runtimeApiBaseUrl: runtimeApiBaseUrl,
        storedApiBaseUrl: storedApiBaseUrl,
      ) ??
      'http://localhost:3000';
}
